pkg/ebpf/c/tracee.bpf.c

```diff
@@ -4153,18 +4176,26 @@ int uprobe_seq_ops_trigger(struct pt_regs *ctx)
         u64 show_addr = (u64) READ_KERN(seq_ops->show);
         if (show_addr == 0)
             return 0;
+        if (show_addr >= (u64) stext_addr && show_addr < (u64) etext_addr)
+            show_addr = 0;
 
         u64 start_addr = (u64) READ_KERN(seq_ops->start);
         if (start_addr == 0)
             return 0;
+        if (start_addr >= (u64) stext_addr && start_addr < (u64) etext_addr)
+            start_addr = 0;
 
         u64 next_addr = (u64) READ_KERN(seq_ops->next);
         if (next_addr == 0)
             return 0;
+        if (next_addr >= (u64) stext_addr && next_addr < (u64) etext_addr)
```

Comment:
i think it would be better to create an `inline` helper function for this check. maybe `is_addr_in_text(u64 addr, u64 stext, u64 etext)`.

Reply:
It's a one liner in both cases but I can do it for the readability purpose if you think it helps, but there's not much "real" encapsulation done this way.